domain/filter/shorthand.go

```diff
@@ -93,3 +93,85 @@ func AddAnd(lhs, rhs Expr) Expr {
 		Rhs:     rhs,
 	}
 }
+
+func addOr(lhs, rhs Expr) Expr {
+	if lhs == nil && rhs == nil {
+		return nil
+	}
+	if lhs == nil {
+		return rhs
+	}
+	if rhs == nil {
+		return lhs
+	}
+	return &LogicOpExpr{
+		LogicOp: Or,
+		Lhs:     lhs,
+		Rhs:     rhs,
+	}
+}
+
+// 期間中に始まる
+// time_start >= min AND time_start < max
+//
+// 期間中に終わる
+// time_end >= min AND time_end < max
+//
+// 期間より前に始まり期間より後に終わる
+// time_start < min AND time_end >= max
```

Comment:
これをORで繋ぐことを明記すると良いと思います
```suggestion
// 以下のいずれかを満たす
// - 期間中に始まる (time_start >= min AND time_start < max)
// - 期間中に終わる (time_end >= min AND time_end < max)
// - 期間より前に始まり期間より後に終わる (time_start < min AND time_end >= max)
```